geopm: Rename and add v3.2.1

bgeltz opened the pull request:
<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->
- Rename geopm-service to libgeopmd.
- Rename geopm-runtime to libgeopm.
- These renames align our spack package names with our upstream package names: https://geopm.github.io/install.html
- Add support for geopmopt when @develop.
- stress-ng: add v0.19.04
- py_sdnotify: new package
- py_scikit_optimize: add v0.10.2

bgeltz committed:
Rename geopm-service to libgeopmd

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Rename geopm-runtime to libgeopm

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
geopm: Add v3.2.1

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
py_sdnotify: new package

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
stress_ng: add v0.19.04

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
geopm: Add geopmopt support

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
py-scikit-optimize: Add 0.10.2

- Also update recipe based on archival of old repo.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz commented:
Hi @tldahlgren.  Please review this when you have a moment.  If there's someone else I should request a review from, please let me know.  Thanks!

tldahlgren commented on repos/spack_repo/builtin/packages/py_scikit_optimize/package.py:
```diff
@@ -15,26 +15,30 @@ class PyScikitOptimize(PythonPackage):
-    pypi = "scikit-optimize/scikit-optimize-0.5.2.tar.gz"
-    git = "https://github.com/scikit-optimize/scikit-optimize.git"
+    homepage = "https://scikit-optimize.readthedocs.io"
+    pypi = "scikit-optimize/scikit_optimize-0.10.2.tar.gz"
```
I cannot confirm the sha256s for `0.9.0` or `0.5.2` after this change. 

You'll need to add a `url_for_version` to account for the `PyPI` switch from a dash to `_`. See the docs at https://spack.readthedocs.io/en/latest/package_review_guide.html#url-url-for-version-or-url-equivalent or, more simply, follow the process at https://github.com/spack/spack-packages/blob/1b897a6a4d5dee4d6b08f22317dbcfa25265df0a/repos/spack_repo/builtin/packages/py_sphinx/package.py#L190.

tldahlgren commented on repos/spack_repo/builtin/packages/py_scikit_optimize/package.py:
```diff
@@ -15,26 +15,30 @@ class PyScikitOptimize(PythonPackage):
     version("0.5.2", sha256="1d7657a4b8ef9aa6d81e49b369c677c584e83269f11710557741d3b3f8fa0a75")
     variant("plots", default=True, description="Build with plot support from py-matplotlib")
 
     depends_on("py-setuptools", type="build")
```
It looks like the new version has a lower bound requirement of `py-setuptools@61.2:` (see https://github.com/holgern/scikit-optimize/blob/v0.10.2/pyproject.toml).

tldahlgren replied:
While you're in there, would you mind reversing the order of dependencies such that the newer version constraints for a given dependency are listed first (e.g., for `py-numpy` list the dependency for `@0.10.2:` first, `@0.9:` second, and unbound third)?

tldahlgren reviewed (changes requested):
Confirmed the new version sha256s and homepage of new package. Confirmed dependencies of new versions of python packages are covered.

I cannot confirm the older versions of `py-scikit-optimize` due to the URL change. One solution is to add `url_for_version` to the package.

bgeltz committed:
py-scikit-optimize: Address review feedback

- Add url_for_version to checksum older releases.
- Add py-setuptools lower bound for 0.10.2.
- Reorder dependencies from newest to oldest.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz commented:
@tldahlgren I believe I've addressed all your concerns and this is ready for another round of review.  Please and thanks.

tldahlgren reviewed (dismissed):
I can now confirm the older versions. Thanks!

bgeltz commented:
@tldahlgren This did not auto-merge yesterday, so I just updated via rebase to try to get it through.  Would you please try again to merge this when you have time? Thanks.

bgeltz commented:
@tldahlgren There seems to be an issue with gitlab-ci and the renaming I've done in this PR.  CI has this error:
```
RuntimeError: cannot concretize 'geopm-runtime', since 'geopm-runtime' does not exist
```
https://gitlab.spack.io/spack/spack-packages/-/jobs/19263375

Is there another spot where I need to perform the rename?  Not sure how to resolve this one.

tldahlgren commented:
> @tldahlgren There seems to be an issue with gitlab-ci and the renaming I've done in this PR. CI has this error:
> 
> ```
> RuntimeError: cannot concretize 'geopm-runtime', since 'geopm-runtime' does not exist
> ```
> 
> https://gitlab.spack.io/spack/spack-packages/-/jobs/19263375
> 
> Is there another spot where I need to perform the rename? Not sure how to resolve this one.


You'll want to run `git grep` on the old package names to see where they're used and change them. I recommend also changing the names in stacks that have the package commented out.

For example, th `e4s` CI stack (https://github.com/spack/spack-packages/blob/6036be63b936ab368e9606c90f88935301d67da5/stacks/e4s/spack.yaml#L118) is trying to build `geopm-runtime` so you'll need to change it.  Looks like it is also referenced in `e4s-oneapi`.

bgeltz committed:
Rename GEOPM packages in stacks/

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz commented:
Ok @tldahlgren I think this is good to go now.  Thanks for the pointers!

tldahlgren reviewed (approved):
Thanks!
